### ISSUE TITLE
Merk: TMS projecting wrong

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#3034]
**[Submitted to the original trac issue database at 1.29pm, Saturday, 5th June 2010]**

Hi,

I just built the latest SVN version of Merkaartor 0.16 (r21588 according the title bar).

I am using a localhost TMS server with a GeoTiff converted with gdal2tiles.py. The original GeoTiff is available here:
http://topo.linz.govt.nz/Topo50_raster_images/GeoTIFFTopo50/CI04_GeoTif_1-00.tif
 (136mb ; oh why couldn't they zip it?)

In openlayers & google/yahoo/etc underneath it all looks good so I'm confident the TMS creation went ok.

Set up the URL in the Merk TMS editor, tile size = 256; 0,0 is bottom left; min = 11; max = 15; proj=epsg:900913.

(I tried min=0 as per ticket #2808 but no luck)

The overlay is some distance out, and that distance changes with Merkaartor zoom level... (screenshots attached)

?
thanks,
Hamish

ps- ubuntu 10.04 (in a VM)

**Comment by tomhughes** [comment]
**Author: HamishB**
**[Added to the original trac issue at 2.02pm, Saturday, 5th June 2010]**

loading the source image as a GeoTiff background with the GDAL plugin works perfectly.

**Comment by tomhughes** [comment]
**Author: Koying**
**[Added to the original trac issue at 2.03pm, Saturday, 5th June 2010]**

Seems typically to indicate a wrong min zoom.

Did you try doing -1 or +1 to to the min zoom?

If that do not work, can you make your tiles available thru the net?

**Comment by tomhughes** [comment]
**Author: SamatJain**
**[Added to the original trac issue at 10.15am, Saturday, 19th June 2010]**

I think I can confirm; this only appears to happen when using TMS with "Origin is bottom-left" as the original reporter is using.

I created a set of tiles with gdal2tiles.py. The output is in TMS standard layout (bottom-left origin). They are here:

  http://maps.rhombic.net/NAIP-2009/Dona_Ana_BLO/%1/%2/%3.png

For compatibility with older versions of Merkaartor/Potlatch, I had written a script that renames tiles from bottom-left origin to top-left origin layout (used by Google, OSM, and older versions of Merkaartor). Those tiles are located here:

  http://maps.rhombic.net/NAIP-2009/Dona_Ana/%1/%2/%3.png

If you browse to the area:

  http://www.openstreetmap.org/?lat=32.34828&lon=-106.72795&zoom=16&layers=B000FTF

You'll notice that the first set of bottom-left origin tiles is misaligned, while the second set with a top-left origin is aligned correctly. The zoom level looks fine, it just looks as if bottom-left has an off-by-one error in a calculation somewhere shifting everything down. Setting minimum zoom to 0 has no effect. I've attached a screenshot of the display.

Both tile sets are tile size 256, min zoom 7, max zoom 17, projection EPSG:900913. I am using Merkaartor 0.16.1 from Debian experimental.
